src/web/src/main.js

```diff
@@ -13,10 +14,24 @@ Vue.config.productionTip = false;
 
 Vue.use(VueRouter);
 Vue.use(VueCookies);
+Vue.use(Vuex);
+
+const store = new Vuex.Store({
+  state: {
+    darkMode: false,
+  },
+  mutations: {
+    setDarkMode(state, bool) {
+      state.darkMode = bool;
+      Vue.$cookies.set("darkMode", bool, null, null, null, null, "Strict");
+    },
+  },
+});
 
 Vue.$cookies.config("7d");
 
 new Vue({
   render: (h) => h(App),
   router,
+  store: store,
```

Comment:
js tip: if the key and variable that contains the value have the same name, you can just put the variable name and js will infer the key name. So e.g. right above on line 35, that is equivalent to `router: router`